python/cudf/cudf/options.py

```diff
@@ -150,6 +150,33 @@ def _validator(val):
     return _validator
 
 
+def _cow_validator(val):
+    if get_option("spill") and val:
+        raise ValueError(
+            "Copy-on-write is not supported when spilling is enabled. "
+            "Please set `spill` to `False`"
+        )
+    if val not in {False, True}:
+        raise ValueError(
+            f"{val} is not a valid option. Must be one of {{False, True}}."
+        )
+
+
+def _spill_validator(val):
+    try:
+        if get_option("copy_on_write") and val:
+            raise ValueError(
+                "Spilling is not supported when Copy-on-write is enabled. "
+                "Please set `copy_on_write` to `False`"
+            )
+    except KeyError:
```

Comment:
Under what circumstances, after this patch lands, can `get_option("copy_on_write")` raise a `KeyError`; and why are we not concerned about the converse case for `get_option("spill")` ?

Reply:
> Under what circumstances, after this patch lands, can `get_option("copy_on_write")` raise a `KeyError`;

`_spill_validator` is executed on `import cudf` because the option `spill` is registered before `copy_on_write` is registered. Hence we would run into a `KeyError` stating that `copy_on_write` is not a registered option. 

> and why are we not concerned about the converse case for `get_option("spill")` ?

Because of the ordering. If we register `copy_on_write` before `spill` we will need to worry about the converse case.

Reply:
IIUC the option being registered below should indeed guarantee that we won't get a KeyError

Reply:
> IIUC the option being registered below should indeed guarantee that we won't get a KeyError

Validators are called during the registration process. So in practice what's happening is `spill` is first in order to be registered and at this point, `copy_on_write` isn't and the `_spill_validator` will get a `KeyError` for the first time it is registering the option.